conntrack.go

```diff
@@ -380,7 +381,44 @@ func (nfct *Nfct) register(ctx context.Context, t Table, groups NetlinkGroup, fi
 					nfct.logger.Printf("could not parse received message: %v", err)
 					continue
 				}
-				if ret := fn(c); ret != 0 {
+				// Attempt to detect the table of the received event
+				var nlTable Table
+				msgTypeTable := (msg.Header.Type&0x300)>>8
+				switch msgTypeTable {
+				case unix.NFNL_SUBSYS_CTNETLINK:
+					nlTable = Conntrack
+				case unix.NFNL_SUBSYS_CTNETLINK_EXP:
+					nlTable = Expected
+				default:
+					nfct.logger.Printf("Unable to detect the table of the received message. Unknown table: %v", msgTypeTable)
+				}
```

Comment:
Why not directly pass the value received from the kernel?
```suggestion
				msgTypeTable := Table((msg.Header.Type&0x300)>>8)
```

Reply:
I haven't verified that the integer values match, but if they do, then yeah, we can cast from integer to Table type. We lose the unknown table check though.

Reply:
your default case breaks in environments where other tables are present. so would favor just passing this value.